### PR TITLE
Slack notification, use semantic changes

### PR DESCRIPTION
**Pull request by sdirosa**
Uses the new space revision semantic changes API to build a detailed summary of what changed, when receiving a `space_content_updated` event callback, and use that for the Slack notification text.

### CODE REVIEW

File: integrations/slack/src/index.ts [hunk 1]
```diff
@@ -26,8 +27,115 @@ const handleSpaceContentUpdated: EventCallback<
...
+                break;
+            case RevisionSemanticChangeType.FileDeleted:
+                deletedFiles.push(change.file.name);
+                break;
+            default:
+                break;
+        }
+    });
+
+    let notificationText = `Content of *${space.title || 'Space'}* has been updated.`;
```
Comment (sdirosa) [inline integrations/slack/src/index.ts:99]: Samy suggested to remove links as we don't know whether the recipient of the notification has access to the space or not. Up to discussion. Easy to reintroduce if we prefer that.

File: integrations/slack/src/index.ts [hunk 2]
```diff
@@ -26,8 +27,117 @@ const handleSpaceContentUpdated: EventCallback<
...
+                deletedFiles.push(change.file.name);
+                break;
+            default:
+                break;
+        }
+    });
+
+    let notificationText = `Content of *<${space.urls.app}|${
+        space.title || 'Space'
+    }>* has been updated.`;
```
Comment (sdirosa) [inline integrations/slack/src/index.ts:101]: A note here. Samy pointed out that linking to spaces may be a issue for some recipients as they may actually not have access to them. I left it as-is for now, but we could just print the space title here, without linking it.

Comment (scazan) [reply integrations/slack/src/index.ts:101]: I'm wondering if there is any downside to them not having access when they click through. Won't it indicate that they don't have access (prompting them to either ask and admin or informing them that they can't actually look at it)? Feels odd that there is no link there when I see it in action.

Comment (scazan) [reply integrations/slack/src/index.ts:101]: I'm gonna file a Linear issue for this topic. I don't think it should hold up pushing it out.

Comment (sdirosa) [reply integrations/slack/src/index.ts:101]: I think it's a hard one to solve. We can't determine here who is going to receive the notification. I think from a UX perspective what you suggest is what makes the most sense. Accepting that some users may receive the notification and get denied and have some nice UI for it.

**Pull request merged**